File: solutions/free-pascal/sum-of-multiples/1/SumOfMultiples.pas
unit SumOfMultiples;

{$mode ObjFPC}{$H+}

interface

function sum(const factors : Array Of Integer; const limit : integer) : integer;

implementation

function sum(const factors : Array Of Integer; const limit : integer) : integer;
var
  number: integer;
  factor: integer;
begin
  result := 0;
  for number := 1 to (limit - 1) do
    for factor in factors do
      if (factor <> 0) and (number mod factor = 0) then
        begin
        result += number;
        break; 
        end;
end;

end.

File: solutions/euphoria/sum-of-multiples/1/sum-of-multiples.ex
include std/math.e

public function sum_of_multiples(sequence f, integer lim)
  integer res = 0
  f += lim * (f = 0)
  for i = 1 to lim - 1 do
    if sum(mod(i, f) = 0) > 0 then
      res += i
    end if
  end for
  return res
end function